airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -220,6 +220,10 @@ private AirbyteFileOffsetBackingStore initializeState(JdbcStateManager stateMana
                                                                              JdbcStateManager stateManager,
                                                                              Instant emittedAt) {
     if (isCdc(config)) {
+      // todo (cgardens) - this is a hack to make sure this flag get set properly. when we consolidate our
+      // CDC logic, we should figure out a way to add this more ergonomically into the AbstractJdbcSource.
+      stateManager.setIsCdc(true);
```

Comment:
This is going to break syncs. In `airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/JdbcStateManager.java` in the constructor we do 
```
 public JdbcStateManager(JdbcState serialized, ConfiguredAirbyteCatalog catalog) {
    this.cdcStateManager = new JdbcCdcStateManager(serialized.getCdcState());
    this.isCdc = serialized.getCdc();
    if (serialized.getCdc() == null) {
      this.isCdc = false;
    }

    pairToCursorInfo =
        new ImmutableMap.Builder<AirbyteStreamNameNamespacePair, CursorInfo>().putAll(createCursorInfoMap(serialized, catalog)).build();
  }
```

Which means `cdc` is already set to false when its null. Now in the `setIsCdc` method we have a check
 
```
  public void setIsCdc(boolean isCdc) {
    if (this.isCdc == null) {
      this.isCdc = isCdc;
    } else {
      Preconditions.checkState(this.isCdc == isCdc, "attempt to set cdc to {}, but is already set to {}.", isCdc, this.isCdc);
    }
  }
```
which would throw the error `java.lang.IllegalStateException: attempt to set cdc to {}, but is already set to {}. [true, false]`